cmd/jujud/agent/introspection.go

```diff
@@ -77,11 +77,11 @@ func startIntrospection(cfg introspectionConfig) error {
 	return nil
 }
 
-// newPrometheusRegistry returns a new prometheus.Registry with
+// NewPrometheusRegistry returns a new prometheus.Registry with
 // the Go and process metric collectors registered. This registry
 // is exposed by the introspection abstract domain socket on all
 // Linux agents.
-func newPrometheusRegistry() (*prometheus.Registry, error) {
+func NewPrometheusRegistry() (*prometheus.Registry, error) {
```

Comment:
I feel this should go somewhere else too. Perhaps something like `cmd/jujud/agent/metrics`.